packages/ptokens-swap/src/ptokens-swap.ts

```diff
@@ -16,20 +18,34 @@ export class pTokensSwap {
   private _destinationAssets: DestinationInfo[]
   private _amount: BigNumber
   private _controller: AbortController
+  private _interimHubAddress: `0x${string}`
+  private _interimNetworkId: NetworkId
+  private _interimProvider: pTokensEvmProvider
 
   /**
    * Create and initialize a pTokensSwap object. pTokensSwap object shall be created using a pTokensSwapBuilder object.
    * @param sourceAsset - The pTokensAsset that will be the source asset for the swap.
    * @param destinationAssets - The pTokensAsset array that will be destination assets for the swap.
    * @param amount - The amount of source asset that will be swapped.
+   * @param interimNetworkId - The NetworkId of the InterimChain.
    */
-  constructor(sourceAsset: pTokensAsset, destinationAssets: DestinationInfo[], amount: BigNumber.Value) {
+  constructor(
+    sourceAsset: pTokensAsset,
+    destinationAssets: DestinationInfo[],
+    amount: BigNumber.Value,
+    interimHubAddress: `0x${string}`,
+    interimNetworkId = INTERIM_CHAIN_NETWORK_ID,
+    interimProvider: pTokensEvmProvider,
+  ) {
     this._sourceAsset = sourceAsset
     if (destinationAssets.length !== 1) throw new Error('There must be one and only one destination asset')
     this._destinationAssets = destinationAssets
     this._amount = BigNumber(amount)
     this._controller = new AbortController()
-    if (!this.isAmountSufficient()) throw new Error('Insufficient amount to cover fees')
+    this._interimHubAddress = interimHubAddress
+    this._interimNetworkId = interimNetworkId
+    this._interimProvider = interimProvider
+    // if (!this.isAmountSufficient()) throw new Error('Insufficient amount to cover fees')
```

Comment:
Why commenting this? the calculation is already aligned with the latest hub code.